### PR TITLE
surround-function improvement for lisp modes

### PR DESCRIPTION
**Pull request by Dickby**
fix a byte-compile warning too.

### CODE REVIEW

**Comment by Dickby** [comment]
the list of lisp modes is copied from smart-parens, I am no expert for lisp, so i'm not shure if this change is the right one for all of those modes.

File: evil-surround.el [hunk 1]
```diff
@@ -157,7 +186,7 @@ See also `evil-surround-inner-overlay'."
         (while (looking-at regexp) (forward-char))
         (evil-set-range-beginning range (point))
         (goto-char (evil-range-end range))
-        (while (looking-back regexp) (backward-char))
+        (while (looking-back regexp nil) (backward-char))
```
Comment (ninrod) [inline evil-surround.el]: what is the diffence? Isn't limit nil by default, since is optional?

File: evil-surround.el [hunk 2]
```diff
@@ -105,8 +133,9 @@ Each item is of the form (OPERATOR . OPERATION)."
 (defun evil-surround-function ()
   "Read a functionname from the minibuffer and wrap selection in function call"
   (let ((fname (evil-surround-read-from-minibuffer "" "")))
-    (cons (format "%s(" (or fname ""))
-          ")")))
+    (cons (format (if (member major-mode evil-surround-lisp-modes)
+		      "(%s " "%s(")
```
Comment (ninrod) [inline evil-surround.el]: I see what you are doing here. I like it. Adding a test would greatly improve your PR.

File: evil-surround.el (hunk 1 above)
Comment (Dickby) [reply evil-surround.el]: Weird, and i get a byte-compile warning, but the function definition sais that LIMIT is optional.
I use Emacs 27.0.50. 
Don't you get that warning?

File: evil-surround.el (hunk 2 above)
Comment (Dickby) [reply evil-surround.el]: please review the test, i'm not shure if it's done right.

File: evil-surround.el (hunk 1 above)
Comment (ninrod) [reply evil-surround.el]: I use emacs 25.3. this is compile warning is really weird.

File: test/evil-surround-test.el [hunk 3]
```diff
@@ -55,10 +55,21 @@
       :visual-end nil
       "argument1 argument2"
       (turn-on-evil-surround-mode)
+      (c-mode)
```
Comment (ninrod) [inline test/evil-surround-test.el]: Why did you activate `c-mode` in this test here?

File: evil-surround.el (hunk 2 above)
Comment (ninrod) [reply evil-surround.el]: I think the test is fine. not sure about the c-mode activation.

File: test/evil-surround-test.el (hunk 3 above)
Comment (Dickby) [reply test/evil-surround-test.el]: I wasn't shure what the default mode is, and to make shure that it's non of the lispy modes, i chose c-mode very basic and present everywhere. Feel free to chane it though.

Comment (ninrod) [reply test/evil-surround-test.el]: That's ok.

**Comment by ninrod** [comment]
Looks good to me! thanks @Dickby.

**Comment by ninrod** [comment]
Hi @Dickby, sorry I had to revert the commit because it breaks a functionality: the repeat on function surround. 

the test suite did not warned us about the failure because `make test` was giving us 0 upon failure. 

see the discussion [here](https://github.com/emacs-evil/evil-surround/pull/135#issuecomment-431175828) on PR #135

**Pull request merged**